internal/local/structure_v2.go

```diff
@@ -331,3 +327,204 @@ func parseSync(rootDir string) (*SyncStructure, error) {
 	}
 	return &SyncStructure{config}, nil
 }
+
+// ConfigData marshals the config data out to JSON
+func (a AppDataV2) ConfigData() ([]byte, error) {
+	temp := &struct {
+		ConfigVersion         realm.AppConfigVersion `json:"config_version"`
+		ID                    string                 `json:"app_id,omitempty"`
+		Name                  string                 `json:"name,omitempty"`
+		Location              realm.Location         `json:"location,omitempty"`
+		DeploymentModel       realm.DeploymentModel  `json:"deployment_model,omitempty"`
+		Environment           string                 `json:"environment,omitempty"`
+		AllowedRequestOrigins []string               `json:"allowed_request_origins,omitempty"`
+	}{
+		ConfigVersion:         a.ConfigVersion(),
+		ID:                    a.ID(),
+		Name:                  a.Name(),
+		Location:              a.Location(),
+		DeploymentModel:       a.DeploymentModel(),
+		Environment:           a.Environment,
+		AllowedRequestOrigins: a.AllowedRequestOrigins,
+	}
+	return MarshalJSON(temp)
+}
+
+// WriteData will write the local Realm app data to disk
+func (a AppDataV2) WriteData(rootDir string) error {
+	if err := writeSecrets(rootDir, a.Secrets); err != nil {
+		return err
+	}
+	if err := writeEnvironments(rootDir, a.Environments); err != nil {
+		return err
+	}
+	if err := writeValues(rootDir, a.Values); err != nil {
+		return err
+	}
+	graphQL := a.GraphQL
+	if a.GraphQL == nil {
+		graphQL = &GraphQLStructure{}
+	}
+	if err := writeGraphQL(rootDir, *graphQL); err != nil {
```

Comment:
[question] does this _always_ want to be written (at least the directory)?  if so, then i was probably mistaken in making this field a pointer to `GraphQLStructure` (and we could remove that).

if not, we should do just do the rare double-nesting here:
```go
if a.GraphQL != nil {
  if err := writeGraphQL(rootDir, *a.GraphQL); err != nil { return err }
}
```

Reply:
I believe so.

Reply:
But rather not change this before our beta release.

Reply:
don't disagree, so let's roll with the wrapping in `!= nil` then